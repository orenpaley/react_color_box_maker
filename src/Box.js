const Box = ({ width, height, backgroundColor, remove, id }) => {
  /** Delete num via parent fn */
  function handleRemove(e) {
    remove(e.target.parentElement);
  }
  return (
    <div
      className="Box"
      id={id}
      style={{
        width: `${width}px`,
        height: `${height}px`,
        border: "2px black solid",
        backgroundColor: backgroundColor,
      }}
    >
      <button onClick={handleRemove}>X</button>
    </div>
  );
};

export default Box;
