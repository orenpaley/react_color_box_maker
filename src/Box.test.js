import Box from "./Box";

import { render } from "@testing-library/react";

// test("renders learn react link", () => {
//   render(<BoxList />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

it("renders without crashing", () => {
  render(<Box id={1} key={1} height={50} width={50} backgroundColor={"red"} />);
});
it("matches snapshot", function () {
  const { asFragment } = render(
    <Box id={1} key={1} height={50} width={50} backgroundColor={"red"} />
  );
  expect(asFragment()).toMatchSnapshot();
});
