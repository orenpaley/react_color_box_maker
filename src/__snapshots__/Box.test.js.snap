// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    id="1"
    style="width: 50px; height: 50px; border: 2px solid black; background-color: red;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
